### ISSUE TITLE
Feature request: "Facade" Segment

### ISSUE THREAD

**Issue by lsuss** [aws/aws-xray-sdk-go#16]
Hi,

In my use-case I need to create a "facade" segment to mimic an existing Segment and send subsegment (and only subsegment) to X-Ray.
There is an equivalent in the nodejs package [aws-cray-sdk-core](https://www.npmjs.com/package/aws-xray-sdk-core)

```js
var facadeSegment = function facadeSegment() {
  var segment = new Segment('facade');
  var whitelistFcn = ['addNewSubsegment', 'addSubsegment', 'removeSubsegment', 'toString'];
  var silentFcn = ['incrementCounter', 'decrementCounter', 'isClosed', 'close', 'format', 'flush'];
  //...
  return segment;
}
```

**Comment by luluzhao** [comment]
@lsuss , thank you for reaching out to us. Unfortunately, we haven't implemented similar method in our Go SDK so far. However, we gonna start writing the method and will be added into upcoming release. I will let you know about it when it released.

**Comment by lsuss** [comment]
@luluzhao thank you for your response. Good to hear it's a work in progress. I would be happy to help by testing it as soon as you make it available :+1:

**Comment by luluzhao** [comment]
@lsuss , in our new released version `v1.0.0-rc.1`, we support the method for creating an facade segment and the method called `BeginFacadeSegment`. We'd like you to try it and let me know if you any questions. Thank you for your patience.

**Comment by nzoschke** [comment]
This is working perfectly for me. Thank you!
